plugins/videobasedtracker/UVCImageSource.cpp

```diff
@@ -192,35 +199,31 @@ namespace vbtracker {
     cv::Size UVCImageSource::resolution() const { return resolution_; }
 
     void UVCImageSource::retrieveColor(cv::Mat &color) {
-        uvc_frame_t *current_frame;
-
-        {
-            // Grab a frame from the queue, but don't keep the queue locked!
-            std::lock_guard<std::mutex> lock(mutex_);
-
-            if (frames_.empty()) {
-                throw std::runtime_error("Error: There's no frames available.");
-            }
-
-            current_frame = frames_.front();
-            frames_.pop();
+        // Grab a frame from the queue, but don't keep the queue locked!
+        std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
With my previous Frame_ptr implementation I needed to RAII it, which meant I couldn't close the scope as I created the Frame_ptr in the same scope, but moving to the stateless implementation will fix this.